python/ray/tests/test_basic_5.py

```diff
@@ -118,5 +119,41 @@ def test_internal_kv(ray_start_regular):
         kv._internal_kv_list("@namespace_abc", namespace="n")
 
 
+def test_run_on_all_workers(ray_start_regular):
+    # This test is to ensure run_function_on_all_workers are executed
+    # on all workers.
+    @ray.remote
+    class Actor:
+        def __init__(self):
+            self.jobs = []
+
+        def record(self, job_id=None):
+            if job_id is not None:
+                self.jobs.append(job_id)
+            return self.jobs
+
+    a = Actor.options(name="recorder", namespace="n").remote()  # noqa: F841
+    driver_script = """
+import ray
+from pathlib import Path
+
+def init_func(worker_info):
+    a = ray.get_actor("recorder", namespace="n")
+    a.record.remote(worker_info['worker'].worker_id)
+
+ray.worker.global_worker.run_function_on_all_workers(init_func)
+ray.init(address='auto')
+@ray.remote
+def ready():
+    a = ray.get_actor("recorder", namespace="n")
+    assert ray.worker.global_worker.worker_id in ray.get(a.record.remote())
```

Comment:
> What was the symptom of sometimes missing functions? If I understand, it was missing the job isolation prefix, so would we never import any of them?

@ericl  yes, we'll never import any of them. So, this assertion will fail without this PR.

Reply:
I see, is run_function_on_all_workers even an API that we support though? What uses this / would break?

Reply:
(Is this something that should block the release?)

Reply:
I think so. It's not a public API, but we are using them here

https://sourcegraph.com/github.com/ray-project/ray/-/blob/python/ray/worker.py?L1578-1581

It'll be used to change the working dir, so for runtime env, I think it'll fail in some cases. I can try to get a script failure.